src/main/java/org/openrefine/extensions/commons/utils/CommonsImportingController.java

```diff
@@ -353,34 +346,35 @@ public List<Object> getNextRowOfCells() throws IOException {
                                 .addQueryParameter("cmcontinue", cmcontinue).build();
                         getFiles(urlContinue);
                         indexRow = 0;
-                        if (mIdsColumn.contentEquals("true")) {
-                            mIdIndex = 0;
-                        }
                     }
                 }
 
                 if (indexRow < files.size()) {
                     rowsOfCells = new ArrayList<>();
-                    rowsOfCells.add(files.get(indexRow++).findValue("title").asText());
+                    rowsOfCells.add(files.get(indexRow).findValue("title").asText());
 
                     if ((categoriesColumn.contentEquals("true")) && (mIdsColumn.contentEquals("true"))) {
-                        //rowsOfCells.add(getCategoriesColumn(files.get(categoriesColumnIndex++).findValue("title").asText()));
-                        rowsOfCells.add("M" + files.get(mIdIndex++).findValue("pageid").asText());
+                        rowsOfCells.add(getCategoriesColumn(files.get(indexRow).findValue("title").asText()));
+                        rowsOfCells.add("M" + files.get(indexRow).findValue("pageid").asText());
+                        indexRow++;
 
                         return rowsOfCells;
 
                     } else if (categoriesColumn.contentEquals("true")) {
                         //FIXME
-                        rowsOfCells.add(getCategoriesColumn(files.get(categoriesColumnIndex++).findValue("title").asText()));
+                        rowsOfCells.add(getCategoriesColumn(files.get(indexRow).findValue("title").asText()));
+                        indexRow++;
 
                         return rowsOfCells;
 
                     } else if (mIdsColumn.contentEquals("true")) {
-                        rowsOfCells.add("M" + files.get(mIdIndex++).findValue("pageid").asText());
+                        rowsOfCells.add("M" + files.get(indexRow).findValue("pageid").asText());
+                        indexRow++;
 
                         return rowsOfCells;
```

Comment:
Because all of your `if` / `else if` / `else` blocks end with `indexRow++; return rowsOfCells` you could move all those instructions outside those blocks, after the `else` block: that would make your code a little more concise.